filters/cropbyfocalpoint.go

```diff
@@ -0,0 +1,130 @@
+package filters
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando-stups/skrop/parse"
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/h2non/bimg.v1"
+	"strconv"
+)
+
+// CropByFocalPointName is the name of the filter
+const CropByFocalPointName = "cropByFocalPoint"
+
+type cropByFocalPoint struct {
+	targetX     float64
+	targetY     float64
+	aspectRatio float64
+}
+
+// NewCropByFocalPoint creates a new filter of this type
+func NewCropByFocalPoint() filters.Spec {
+	return &cropByFocalPoint{}
+}
+
+func (f *cropByFocalPoint) Name() string {
+	return CropByFocalPointName
+}
+
+func (f *cropByFocalPoint) CreateOptions(imageContext *ImageFilterContext, filterContext filters.FilterContext) (*bimg.Options, error) {
```

Comment:
The filter context should not be passed here. Instead please extend [this method](https://github.com/zalando-stups/skrop/blob/59b76fe5c1e088ee131d62c37199566f2cb81305/filters/imagefilter.go#L93) to add the required information from the `FilterContext` to the `ImageFilterContext`